packages/exo/test/test-legacy-guard-tolerance.js

```diff
@@ -0,0 +1,140 @@
+// @ts-check
+// eslint-disable-next-line import/order
+import { test } from './prepare-test-env-ava.js';
+
+import { passStyleOf } from '@endo/pass-style';
+import {
+  M,
+  getAwaitArgGuardPayload,
+  getInterfaceGuardPayload,
+  getMethodGuardPayload,
+} from '@endo/patterns';
+import {
+  makeLegacyAwaitArgGuard,
+  makeLegacyMethodGuard,
+  makeLegacyInterfaceGuard,
+} from './make-legacy-guards.js';
+import { makeExo } from '../src/exo-makers.js';
+
+test('legacy guard tolerance', async t => {
+  const aag = M.await(88);
+  const laag = makeLegacyAwaitArgGuard({
+    argGuard: 88,
+  });
+  const mg1 = M.callWhen(77, aag).returns(M.any());
+  const mg2 = M.callWhen(77, laag).returns(M.any());
+  const lmg = makeLegacyMethodGuard({
+    callKind: 'async',
+    argGuards: [77, laag],
+    returnGuard: M.any(),
+  });
+  const ig1 = M.interface('Foo', {
+    mg1,
+    mg2,
+  });
+  const lig = makeLegacyInterfaceGuard({
+    interfaceName: 'Foo',
+    methodGuards: {
+      mg1,
+      mg2,
+      // @ts-expect-error Legacy adaptor can be ill typed
+      lmg,
+    },
+  });
+  t.is(passStyleOf(laag), 'copyRecord');
+  t.is(passStyleOf(lmg), 'copyRecord');
+  t.is(passStyleOf(lig), 'copyRecord');
+
+  t.deepEqual(getAwaitArgGuardPayload(aag), {
+    argGuard: 88,
+  });
+  // @ts-expect-error Legacy adaptor can be ill typed
+  t.throws(() => getAwaitArgGuardPayload(laag), {
+    message:
+      'awaitArgGuard: copyRecord {"argGuard":88,"klass":"awaitArg"} - Must be a guard:awaitArgGuard',
+  });
+
+  t.deepEqual(getMethodGuardPayload(mg1), {
+    callKind: 'async',
+    argGuards: [77, aag],
+    optionalArgGuards: undefined,
+    restArgGuard: undefined,
+    returnGuard: M.any(),
+  });
+  t.deepEqual(getMethodGuardPayload(mg2), {
+    callKind: 'async',
+    argGuards: [77, laag],
+    optionalArgGuards: undefined,
+    restArgGuard: undefined,
+    returnGuard: M.any(),
+  });
+  // @ts-expect-error Legacy adaptor can be ill typed
+  t.throws(() => getMethodGuardPayload(lmg), {
+    message:
+      'methodGuard: copyRecord {"argGuards":[77,{"argGuard":88,"klass":"awaitArg"}],"callKind":"async","klass":"methodGuard","returnGuard":"[match:any]"} - Must be a guard:methodGuard',
+  });
+
+  t.deepEqual(getInterfaceGuardPayload(ig1), {
+    interfaceName: 'Foo',
+    methodGuards: {
+      mg1,
+      mg2,
+    },
+    defaultGuards: undefined,
+  });
+  // would have been ig2 is it could be made
```

Comment:
Hard to parse. Is this what was meant?
```suggestion
  // would have been ig2 if it could be made
```

Reply:
yes, thanks. Done.